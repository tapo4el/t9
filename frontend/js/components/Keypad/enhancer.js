import { connect } from 'react-redux';

import { addSymbol, removeSymbol } from '../../actions';

const map = {
    2: 'a', 3: 'd', 4: 'g', 5: 'j', 6: 'm', 7: 'p', 8: 't', 9: 'w',
};

const mapDispatchToProps = dispatch => ({
    addSymbol: payload => dispatch(addSymbol({
        number: payload,
        symbol: map[payload],
    })),
    removeSymbol: () => dispatch(removeSymbol()),
});

export default connect(null, mapDispatchToProps);
