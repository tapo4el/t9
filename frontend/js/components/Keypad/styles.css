.t9-app .keypad {
    flex:1 1;
}
.t9-app .buttons {
    display: flex;
    width: 250px;
}
