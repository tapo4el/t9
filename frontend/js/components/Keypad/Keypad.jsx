import React from 'react';
import { func } from 'prop-types';

import Button from '../Button';
import './styles.css';

class Keypad extends React.PureComponent {
    static propTypes = {
        addSymbol: func.isRequired,
        removeSymbol: func.isRequired,
    };

    render() {
        const { addSymbol, removeSymbol } = this.props;
        return (
            <div className="keypad">
                <div className="buttons">
                    <Button onClickHandler={() => {}} value="1" />
                    <Button onClickHandler={addSymbol} value="2" text="abc" />
                    <Button onClickHandler={addSymbol} value="3" text="def" />
                </div>
                <div className="buttons">
                    <Button onClickHandler={addSymbol} value="4" text="ghi" />
                    <Button onClickHandler={addSymbol} value="5" text="jkl" />
                    <Button onClickHandler={addSymbol} value="6" text="mno" />
                </div>
                <div className="buttons">
                    <Button onClickHandler={addSymbol} value="7" text="pqrs" />
                    <Button onClickHandler={addSymbol} value="8" text="tuv" />
                    <Button onClickHandler={addSymbol} value="9" text="wxyz" />
                </div>
                <div className="buttons">
                    <Button onClickHandler={removeSymbol} value="Remove" />
                </div>
            </div>
        );
    }
}

export default Keypad;
