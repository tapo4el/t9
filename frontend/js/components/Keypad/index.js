import Keypad from './Keypad';
import enhancer from './enhancer';

export default enhancer(Keypad);
