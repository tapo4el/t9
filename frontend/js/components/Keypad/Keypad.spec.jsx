import React from 'react';
import renderer from 'react-test-renderer';

import Keypad from './Keypad';

describe('Keypad', () => {
    it ('should render correctly', () => {
        const component = renderer.create(
            <Keypad addSymbol={()=>{}} removeSymbol={()=>{}}/>
        );

        expect(component.toJSON()).toMatchSnapshot();
    });
});
