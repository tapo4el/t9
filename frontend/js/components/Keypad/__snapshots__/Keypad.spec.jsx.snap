// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Keypad should render correctly 1`] = `
<div
  className="keypad"
>
  <div
    className="buttons"
  >
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      1
      <div />
    </button>
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      2
      <div>
        abc
      </div>
    </button>
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      3
      <div>
        def
      </div>
    </button>
  </div>
  <div
    className="buttons"
  >
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      4
      <div>
        ghi
      </div>
    </button>
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      5
      <div>
        jkl
      </div>
    </button>
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      6
      <div>
        mno
      </div>
    </button>
  </div>
  <div
    className="buttons"
  >
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      7
      <div>
        pqrs
      </div>
    </button>
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      8
      <div>
        tuv
      </div>
    </button>
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      9
      <div>
        wxyz
      </div>
    </button>
  </div>
  <div
    className="buttons"
  >
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      Remove
      <div />
    </button>
  </div>
</div>
`;
